src/certificate.rs

```diff
@@ -206,6 +204,12 @@ impl X509CertificateParser {
     }
 }
 
+impl Default for X509CertificateParser {
+    fn default() -> Self {
+        X509CertificateParser::new()
+    }
```

Comment:
(just a comment) I know clippy prefers the `Default` trait (especially when `new()` takes no argument like here), however it does not allow declaring the method `const` nor `inline` (which I do use a lot).
So, adding an implementation for `Default`, but *not* replacing `new`, looks good to me